tests/lax_numpy_test.py

```diff
@@ -1292,19 +1292,36 @@ def args_maker():
     self._CompileAndCheck(jnp_fun, args_maker)
 
   @parameterized.named_parameters(jtu.cases_from_list(
-      {"testcase_name": "_shape=[{}]_axis={}_repeats={}".format(
-          jtu.format_shape_dtype_string(shape, dtype), axis, repeats),
+      {"testcase_name": "_shape=[{}]_axis={}_repeats={}_fixed_size={}".format(
+          jtu.format_shape_dtype_string(shape, dtype),
+          axis, repeats, fixed_size),
        "axis": axis, "shape": shape, "dtype": dtype, "repeats": repeats,
-       "rng_factory": jtu.rand_default}
+       "rng_factory": jtu.rand_default, 'fixed_size': fixed_size}
       for repeats in [0, 1, 2]
       for shape, dtype in _shape_and_dtypes(all_shapes, default_dtypes)
-      for axis in [None] + list(range(-len(shape), max(1, len(shape))))))
-  def testRepeat(self, axis, shape, dtype, repeats, rng_factory):
+      for axis in [None] + list(range(-len(shape), max(1, len(shape))))
+      for fixed_size in [True, False]))
+  def testRepeat(self, axis, shape, dtype, repeats, rng_factory, fixed_size):
     rng = rng_factory(self.rng())
     np_fun = lambda arg: np.repeat(arg, repeats=repeats, axis=axis)
     np_fun = _promote_like_jnp(np_fun)
-    jnp_fun = lambda arg: jnp.repeat(arg, repeats=repeats, axis=axis)
+    if fixed_size:
+      # Calculate expected size of the repeated axis.
+      if axis is None:
+        axis = 0
+      if jnp.ndim(shape) == 0:
+        total_repeat_length = repeats
+      else:
+        # Broadcasting the repeats if a scalar value.
+        expected_repeats = jnp.broadcast_to(jnp.ravel(repeats),
+                                            [shape[axis]])
+        # Total size will be num_repeats X axis length.
+        total_repeat_length = jnp.sum(expected_repeats)
+    else:
+      total_repeat_length=None
 
+    jnp_fun = lambda arg: jnp.repeat(arg, repeats=repeats, axis=axis,
+                                     total_repeat_length=total_repeat_length)
     args_maker = lambda: [rng(shape, dtype)]
 
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
```

Comment:
Last thing I would do here is add a ``self._CompileAndCheck`` in the case that `fixed_size` is True, because the function should be compilable in that case.

Reply:
Funily enough, this is always compilable for the current test cases because they only have scalar repeats. I'll add some array repeats.

Reply:
Oh I think I misunderstood - repeats is a constant in a closure in the current use case so that's while it is already compiling! My bad.